app/code/Magento/Catalog/Model/Product/Price/BasePriceStorage.php

```diff
@@ -225,4 +266,35 @@ private function retrieveValidPrices(array $prices)
 
         return $prices;
     }
+
+    /**
+     * If Catalog Price Mode is Website, price needs to be applied to all Store Views in this website.
+     *
+     * @param array $formattedPrices
+     * @return array
+     */
+    private function applyWebsitePrices($formattedPrices)
+    {
+        foreach ($formattedPrices as $price) {
+            if ($price['store_id'] == Store::DEFAULT_STORE_ID) {
+                continue;
+            }
+
+            try {
+                $storeIds = $this->storeManager->getStore($price['store_id'])->getWebsite()->getStoreIds();
+            } catch (NoSuchEntityException $e) {
+                continue;
```

Comment:
I would throw here an exception directly without allowing user to even save any prices